bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/RetryListener.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.io.net.http;
+
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.api.Response;
+import org.eclipse.jetty.client.api.Result;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is an implementation of {@link Response.CompleteListener}, which retries an HTTP Request, if it has failed.
+ *
+ * @author Svilen Valkanov - initial contribution
+ */
+
+public class RetryListener implements Response.CompleteListener {
```

Comment:
@kaikreuzer 
I made it public, because I could not find a replacement for [DefaultHttpMethodRetryHandler](https://hc.apache.org/httpclient-3.x/apidocs/org/apache/commons/httpclient/DefaultHttpMethodRetryHandler.html) from Apache HttpClient 3.x.

The handler is used in [fsinternetaudio](https://github.com/eclipse/smarthome/pull/1864/files#diff-e8aface86649dcb555149ba4b6f0b317L81) as well, and I thought it could it could be used from other bundles as well.

When we do not need it here, I guess I can move it to the fsinternetaudio binding (before that I should fix the onComplete() method of course ) and make it private.